make reset-versions fails on fresh clone

### Steps to Reproduce

1. Create a fresh clone of xamarin-macios
2. Run `make reset-versions`
3. Wait for make to finish cloning LLVM, then watch it fail.

### Expected Behavior

`make` should keep cloning the other dependencies.

### Actual Behavior

`make` fails (see error log below). I think this is due to a dependency error between the LLVM and Mono submodules.

### Environment

```
=== Visual Studio Community 2017 for Mac ===

Version 7.4 (build 1033)
Installation UUID: 2f7490c4-93db-429d-9511-4b9317c93fbc
Runtime:
	Mono 5.8.0.129 (2017-10/835a98d8559) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)

	Package version: 508000129

=== NuGet ===

Version: 4.3.1.4445

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.0.5
	2.0.0
SDK: /usr/local/share/dotnet/sdk/2.1.4/Sdks
SDK Versions:
	2.1.4
	2.0.0
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.8.0/lib/mono/msbuild/15.0/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.1
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Xamarin.Android ===

Not Installed

=== Apple Developer Tools ===

Xcode 9.2 (13772)
Build 9C40b

=== Xamarin.Mac ===

Version: 4.2.0.20 (Visual Studio Community)

=== Xamarin.iOS ===

Version: 11.8.0.20 (Visual Studio Community)
Hash: 1c6f300e
Branch: d15-6
Build date: 2018-02-07 23:42:40-0500

=== Xamarin Inspector ===

Version: 1.4.0
Hash: b3f92f9
Branch: master
Build date: Fri, 19 Jan 2018 22:00:34 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 704001033
Git revision: 053991068f3c88497d3e5cc52fe7fdf57c28b62f
Build date: 2018-02-28 05:02:28-05
Xamarin addins: cc3283953e0af0b8bfab1bdfdf847d9e403c4606
Build lane: monodevelop-lion-d15-6

=== Operating System ===

Mac OS X 10.13.3
Darwin 17.4.0 Darwin Kernel Version 17.4.0
    Sun Dec 17 09:19:54 PST 2017
    root:xnu-4570.41.2~1/RELEASE_X86_64 x86_64

=== Enabled user installed addins ===

EditorConfig 1.2
Eto.Forms Support Addin 2.4.0.0
AddinMaker 1.4.2
Internet of Things (IoT) development (Preview) 7.1
```

### Build Logs

```
*** git submodule update --init --recursive --force -- ./external/llvm
cd /Volumes/Sources/xamarin-macios && git submodule update --init --recursive --force -- ./external/llvm
Submodule path 'external/llvm': checked out 'bdb3a116dbf4a77ef5f6f70d7f845bdaca2b24ca'
echo "DISABLE_ANDROID=1" > ../external/mono/sdks/Make.config
/bin/sh: ../external/mono/sdks/Make.config: No such file or directory
make[1]: *** [../external/mono/sdks/Make.config] Error 1
make: *** [reset-llvm] Error 2
```

yes, I get the same error
```
Sebastiens-MacBook-Pro:git poupou$ mkdir test
Sebastiens-MacBook-Pro:git poupou$ cd test/
Sebastiens-MacBook-Pro:test poupou$ git clone git@github.com:xamarin/xamarin-macios.git
Cloning into 'xamarin-macios'...
remote: Counting objects: 43057, done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 43057 (delta 1), reused 0 (delta 0), pack-reused 43049
Receiving objects: 100% (43057/43057), 75.83 MiB | 2.07 MiB/s, done.
Resolving deltas: 100% (31599/31599), done.
Checking out files: 100% (4706/4706), done.
Sebastiens-MacBook-Pro:test poupou$ cd xamarin-macios/
Sebastiens-MacBook-Pro:xamarin-macios poupou$ make reset-versions
*** git submodule update --init --recursive --force -- ./external/llvm
cd /Users/poupou/git/test/xamarin-macios && git submodule update --init --recursive --force -- ./external/llvm
Submodule 'external/llvm' (git@github.com:mono/llvm.git) registered for path 'external/llvm'
Cloning into '/Users/poupou/git/test/xamarin-macios/external/llvm'...
Submodule path 'external/llvm': checked out 'bdb3a116dbf4a77ef5f6f70d7f845bdaca2b24ca'
echo "DISABLE_ANDROID=1" > ../external/mono/sdks/Make.config
/bin/sh: ../external/mono/sdks/Make.config: No such file or directory
make[1]: *** [../external/mono/sdks/Make.config] Error 1
make: *** [reset-llvm] Error 2
Sebastiens-MacBook-Pro:xamarin-macios poupou$ 
```

The `echo` is done before cloning `mono` so it will fail on a first checkout.

```
864091d33 (Zoltan Varga        2018-03-22 21:25:00 -0400  105) 	echo "DISABLE_ANDROID=1" > $@
```
c.c. @vargaz

--- a/mk/versions.mk
+++ b/mk/versions.mk
@@ -84,7 +84,7 @@ reset-mono::
 	$(Q) rm -f $(TOP)/.stamp-build* $(MONO_PATH)/configure
 
 reset-llvm::
-	$(Q) $(MAKE) -C $(TOP)/builds clean-llvm
+	$(Q) if test -d $(MONO_PATH); then $(MAKE) -C $(TOP)/builds clean-llvm; fi
 
 reset-fsharp::
 	$(Q) cd $(FSHARP_PATH) && git clean -xffdq

